Packs/KnowBe4_KMSAT/Integrations/KnowBe4KMSATEventCollector/KnowBe4KMSATEventCollector_test.py

```diff
@@ -0,0 +1,156 @@
+import datetime
+import pytest
+import json
+import io
+from CommonServerPython import CommandResults, parse_date_string
+from KnowBe4KMSATEventCollector import Client
+
+
+def util_load_json(path):
+    with io.open(path, mode='r', encoding='utf-8') as f:
+        return json.loads(f.read())
+
+
+MOCK_ENTRY = util_load_json('test_data/mock_event.json')
+BASE_URL = 'https://api.events.knowbe4.com'
+
+
+def test_test_module(requests_mock):
+    """
+    Given:
+        - test-module call
+    When:
+        - A response with an OK status_code is retrieved from the API call.
+    Then:
+        - Make sure 'ok' is returned.
+    """
+    from KnowBe4KMSATEventCollector import test_module
+
+    requests_mock.get(
+        f'{BASE_URL}/events',
+        json=MOCK_ENTRY
+    )
+    assert test_module(Client(base_url=BASE_URL)) == 'ok'
+
+
+@pytest.mark.parametrize('last_run, mock_item, expected_last_run, expected_fetched_events', [
+    ({'latest_event_time': "2022-08-05T10:05:03.000Z"}, MOCK_ENTRY,
+     {'latest_event_time': "2022-08-09T10:05:13.890Z"}, MOCK_ENTRY.get('data', [])[0:-1]),
+    ({'latest_event_time': parse_date_string("2022-08-05T10:05:03.000Z")}, MOCK_ENTRY,
+     {'latest_event_time': "2022-08-09T10:05:13.890Z"}, MOCK_ENTRY.get('data', [])[0:-1])])
+def test_fetch_events(requests_mock, last_run, mock_item, expected_last_run, expected_fetched_events):
+    """
+    Given:
+        - last_run marker and a mock with 2 events that occurred after the last run and 1 that occurred before.
+        - case 1: The last run object is a string (an example of last run that was fetched from previous interval).
+        - case 2: The last run object is a datetime object (an example of last run that was created by a default datetime.now()
+                  in the previous interval due to no fetched events).
+    When:
+        - Calling fetch events.
+    Then:
+        - Make sure 2 events returned.
+        - Verify the new lastRun is calculated correctly.
+        - Verify that the function handle the different last_run types correctly.
+    """
+    from KnowBe4KMSATEventCollector import fetch_events
+
+    requests_mock.get(
+        f'{BASE_URL}/events',
+        json=mock_item
+    )
+    events, new_last_run = fetch_events(Client(base_url=BASE_URL), last_run=last_run)
+    expected_last_run == new_last_run
+    assert events == expected_fetched_events
+    assert len(events) == 2
```

Comment:
```suggestion
    assert len(events) == len(expected_fetched_events)
```
but this whole line is probably redundant as its checked in the line before